frontends/web/src/routes/account/summary/summarybalance.tsx

```diff
@@ -0,0 +1,217 @@
+/**
+ * Copyright 2023 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useTranslation } from 'react-i18next';
+import * as accountApi from '../../../api/account';
+import { Balances, SyncStatus } from './accountssummary';
+import { route } from '../../../utils/route';
+import Logo from '../../../components/icon/logo';
+import { Amount } from '../../../components/amount/amount';
+import Spinner from '../../../components/spinner/ascii';
+import { FiatConversion } from '../../../components/rates/rates';
+import { Skeleton } from '../../../components/skeleton/skeleton';
+import style from './accountssummary.module.css';
+
+type TProps = {
+  accounts: accountApi.IAccount[],
+  summaryData?: accountApi.ISummary,
+  totalBalancePerCoin?: accountApi.ITotalBalance,
+  balances?: Balances,
+  syncStatus?: SyncStatus,
+}
+
+type TBalanceRow = {
+    code: accountApi.AccountCode;
+    name: string;
+    balance?: accountApi.IAmount;
+    coinUnit: string;
+    coinCode: accountApi.CoinCode;
+    coinName: string;
+};
+
+type TAccountCoinMap = {
+    [code in accountApi.CoinCode]: accountApi.IAccount[];
+};
+
+export function SummaryBalance ({
+  accounts,
+  summaryData,
+  totalBalancePerCoin,
+  balances,
+  syncStatus,
+}: TProps) {
+  const { t } = useTranslation();
+
+  const getAccountsPerCoin = () => {
+    return accounts.reduce((accountPerCoin, account) => {
+      accountPerCoin[account.coinCode]
+        ? accountPerCoin[account.coinCode].push(account)
+        : accountPerCoin[account.coinCode] = [account];
+      return accountPerCoin;
+    }, {} as TAccountCoinMap);
+  };
+
+  const balanceRow = (
+    { code, name, coinCode }: TBalanceRow,
+  ) => {
+    const balance = balances ? balances[code] : undefined;
```

Comment:
If youre wondering what's reason why we shouldn't declare a react component inside another, it's because doing so (`subTotalRow` inside `SummaryBalance`, for example) is an anti-pattern.

Read: https://stackoverflow.com/a/59776934/9786502
TL;DR: This causes whatever is inside `SummaryBalance` to get re-created every re-render of `SummaryBalance`, which becomes inefficient.

Might be interesting read: https://levelup.gitconnected.com/code-review-avoid-declaring-react-component-inside-parent-component-1768a645f523

Reply:
Good to know! Thanks :)
Moved `balanceRow` and `subTotalRow` into dedicated components. Also simplified a bit `renderAccountSummary` and embedded it directly into the return JSX code.
PTAL :pray: